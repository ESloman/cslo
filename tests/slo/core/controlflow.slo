
println("Should be less than 5");
var x = 5;
if (x < 10) {
    println("less than 10!");
} else {
    println("10 or more!");
}

println("Print up to 20");
while (x < 20) {
    x = x + 1;
    println(x);
}

println("Shoud be more than 10");
if (x < 10) {
    println("less than 10!");
} else {
    println("10 or more!");
}

println("Decrementing");
while (x > 5) {
    x = x - 1;
    println(x);
}

println("Inline for");
for (var i = 0; i < 2; i++) { println(i); }

println("Using global x");
for (; x < 20; x++) {
    println(x);
}
