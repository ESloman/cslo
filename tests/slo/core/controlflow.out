Should be less than 5
less than 10!
Print up to 20
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
Shoud be more than 10
10 or more!
Decrementing
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
Inline for
0
1
Using global x
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
